### DISCUSSION TITLE
WAV plays slow

### DISCUSSION THREAD

**Discussion by dca00** [naudio/NAudio#978] [Q&A] [answered]
Mark,

Using a wave recorder and a WASAPI player, I am able to record 16000 Hz samples and play them back. All I do is capture a buffer and add it to the playback.
But when I try to add to that same player a WAV read from a file by a WAV file reader that WAV plays slow.
Beforehand, I converted that file to 16000, but I also tried 48000, and they play at the same slow rate. What is necessary to play a WAV at the same rate as the samples being captured?

**Comment by markheath** [comment, marked as answer]
The WAV file header must specify the same channel count and sample rate that the audio was captured at. You can't just change that number without also resampling the audio. I suspect there is a mismatch between the WAV header and the capture rate of the samples in the file

**Comment by dca00** [comment]
I thought that Audacity exports consistent data and header. The file plays correctly in all sorts of players, so I am not sure what to think.

**Comment by dca00** [comment]
Mark:

Thanks for your pointer!

Audacity played a dirty trick on me: resampling to 16000 does not actually do anything, even though the track shows 16k after resampling. There is Audacity forum thread on this: the export takes the rate from the bottom left corner, not from the current track's rate.

After having switched the project rate and re-exported the file at 16k, it plays at the correct speed.

But this begs another question: having to convert all effect assets to the voice recording rate does not sound like a sensible approach. Is it possible to resample asset WAVs to the current recording rate, whatever it might be? I saw your article  on MP3 resampling at https://markheath.net/post/how-to-resample-audio-with-naudio but am not sure it applies to WAV as well.

**Comment by Axwabo** [comment]
> Is it possible to resample asset WAVs to the current recording rate

You could fake the input sample rate with a custom `ISampleProvider` (change the wave format) which you could pass to a `WdlResamplingSampleProvider`
